File: src/main/java/com/lessmarkup/interfaces/module/ModuleIntegration.java
package com.lessmarkup.interfaces.module;

import com.lessmarkup.interfaces.data.DataObject;
import com.lessmarkup.interfaces.structure.NodeHandler;
import com.lessmarkup.interfaces.structure.Tuple;
import java.util.Collection;

public interface ModuleIntegration {
    void registerBackgroundJobHandler(BackgroundJobHandler handler);
    boolean doBackgroundJobs();
    void registerEntitySearch(Class<? extends DataObject> type, EntitySearch entitySearch);
    void registerUserPropertyProvider(UserPropertyProvider provider);
    EntitySearch getEntitySearch(Class<? extends DataObject> collectionType);
    Collection<UserProperty> getUserProperties(long userId);
    void registerActionHandler(String name, ModuleActionHandler handler);
    ModuleActionHandler getActionHandler(String name);
}

File: src/main/scala/com/lessmarkup/interfaces/module/ModuleIntegration.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.module

import com.lessmarkup.interfaces.data.DataObject

trait ModuleIntegration {
  def registerBackgroundJobHandler(handler: BackgroundJobHandler)
  def doBackgroundJobs(): Boolean
  def registerEntitySearch(dataType: Class[_ <: DataObject], entitySearch: EntitySearch)
  def registerUserPropertyProvider(provider: UserPropertyProvider)
  def getEntitySearch(collectionType: Class[_ <: DataObject]): EntitySearch
  def getUserProperties(userId: Long): Seq[UserProperty]
  def registerActionHandler(name: String, handler: ModuleActionHandler)
  def getActionHandler(name: String): ModuleActionHandler
}